tests/console/yast2_registration.pm

```diff
@@ -0,0 +1,63 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: check if an unregistered system can be registered and if
+#          enabling and disabling extensions correctly work.
+# Maintainer: Paolo Stivanin <pstivanin@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+
+sub register_system_and_add_extension {
+    wait_screen_change { type_string get_var "SCC_EMAIL" };
+    send_key "tab";
+    wait_screen_change { type_string get_var "SCC_REGCODE" };
+    send_key "alt-n";
+    assert_screen("yast2_registration-ext-mod-selection", timeout => 60);
+    # enable Web and Scripting Module
+    for my $i (0 .. 25) {
+        send_key "down";
+    }
+    wait_screen_change { send_key "spc" };
+    send_key "alt-n";
+    wait_still_screen(stilltime => 7, timeout => 60);
+    if (check_screen "yast2_registration-license-agreement") {
+        wait_screen_change { send_key "alt-a" };
+        send_key "alt-n";
+        wait_still_screen 5;
+    }
+    send_key "alt-a";
+    wait_still_screen 2;
+    send_key "alt-o";
+    wait_still_screen 5;
+    wait_screen_change { send_key "alt-f" };
+}
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    zypper_call "in yast2-registration";
```

Comment:
hm isn't yast-registration is something which should present **always** by default in any SLE system ? and if it is not available we can treat it as a bug ?

Reply:
It's not always present: http://d502.qam.suse.de/tests/1604#step/yast2_registration/2